### DISCUSSION TITLE
Can we set the Validation queue size for gossipsub in rust, just like go-libp2p?

### DISCUSSION THREAD

**Discussion by harsh4786** [libp2p/rust-libp2p#4396] [Q&A]
`ps, err := pubsub.NewGossipSub(ctx, h,
			pubsub.WithValidateQueueSize(P2P_VALIDATE_QUEUE_SIZE),
			pubsub.WithGossipSubParams(gossipParams),
)`

In this snippet the `WithValidateQueueSize` is the param that can be set, I was wondering if it's possible in rust

**Comment by mxinden** [comment]
I don't think rust-libp2p's gossipsub implementation has a validation queue. Please correct me if I am wrong.

**Reply by dariusc93** [reply]
I dont think the rust implementation of gossipsub has one either from my knowledge. Would have to look to see what go-libp2p does with it though and if its something worth having in rust-libp2p if its not already implemented.

**Reply by harsh4786** [reply]
Im trying to port a codebase from go-libp2p to rust-libp2p.
